src/test/java/io/supertokens/test/emailpassword/PasswordHashingTest.java

```diff
@@ -16,11 +16,14 @@
 
 package io.supertokens.test.emailpassword;
 
+import com.lambdaworks.crypto.SCrypt;
```

Comment:
Test cases you need to cover:
- [ ] Argon2 / bcrypt hashing works with all the various types
- [ ] Firebase input hash not meeting right format in API yields a 400 error response (try all combinations of incorrect input)
- [ ] Once firebase password hash is saved, verifying the passwords works.
- [ ] If we are keeping the modifying prefix thing, the flow should work even if the firebase hash doesn't have the prefix.
- [ ] Doing several firebase hashes (similar to how we did for argon2), shouldn't yield the server to crash - this is testing the pool size thing
- [ ] Not providing the firebase signing key, but providing a firebase hash in the import API returns a 400 error.
- [ ] Not providing the firebase signing key, but trying to sign in with a firebase password should yield a 400 error.
- [ ] Providing a wrong firebase signing key and trying to sign in with a firebase password should yield invalid credentials status response. Here the signing key should be a a random string.
- [ ] Store two different firebase hashes with different memcost / rounds etc and make sure that verifying the password against those hashes still works.
- [ ] Providing wrong password during sign in with firebase password should yield invalid credentials status response.
- [ ] If we have stored the firebase password hash is incorrectly stored in the db, and then trying to sign in with the right password should give invalid credentials status response.
- [ ] Jumbling the oder of the rounds, memory, salt separator should still make things work.
- [ ] What happens is salt separator contains a `$` or `$$`, or `$dsf$$` or some other combination of using `$`? Can we still make it work?